azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_sonarcloud.go

```diff
@@ -48,5 +48,5 @@ func flattenServiceEndpointSonarCloud(d *schema.ResourceData, serviceEndpoint *s
 
 	tfhelper.HelpFlattenSecret(d, "token")
 
-	d.Set("token", (*serviceEndpoint.Authorization.Parameters)["username"])
+	d.Set("token", (*serviceEndpoint.Authorization.Parameters)["token"])
```

Comment:
service will not return sensitive data like token/password etc. , this will set the token to empty string. Next time run `terraform apply` the token will updated to empty string which will case the token lost.  The `token` should set from the local config or leave it not updated.

Reply:
Thank you, I have left it out not to be updated for now, although I'm curious how I could set it from the local config.

Reply:
use `d.Get("")`

Reply:
Done